core/src/main/scala/sttp/openai/requests/completions/chat/message/Message.scala

```diff
@@ -0,0 +1,83 @@
+package sttp.openai.requests.completions.chat.message
+
+import sttp.openai.json.SnakePickle
+import sttp.openai.requests.completions.chat.ToolCall
+import ujson._
+
+sealed trait Message
+
+object Message {
+  case class SystemMessage(content: String, name: Option[String] = None) extends Message
+  case class UserMessage(content: Content, name: Option[String] = None) extends Message
+  case class AssistantMessage(content: String, name: Option[String] = None, toolCalls: Option[Seq[ToolCall]] = None) extends Message
```

Comment:
** nitpick ** `Option[Seq[T]]` is sometimes confusing for users, as it is difficult to guess the difference between `None` and `Some(Nil)`. If there's no difference in this case, then maybe we can make it even more convenient for the user and put just a `Seq[T] = Nil` here, which is treated as `None` by the serializer if empty?